tests/ops/integration_tests/saas/test_zendesk_task.py

```diff
@@ -1,347 +1,122 @@
+        base_url = f"https://{secrets['domain']}"
+
+        # user
+        body = {
+            "user": {
+                "name": "Ethyca Test Erasure",
+                "email": zendesk_erasure_identity_email,
+                "verified": "true",
+            }
+        }
+
+        users_response = requests.post(
+            url=f"{base_url}/api/v2/users", auth=auth, json=body
+        )
+        user = users_response.json()["user"]
+        user_id = user["id"]
+
+        # ticket
+        ticket_data = {
+            "ticket": {
+                "comment": {"body": "Test Comment"},
+                "priority": "urgent",
+                "subject": "Test Ticket",
+                "requester_id": user_id,
+                "submitter_id": user_id,
+                "description": "Test Description",
+            }
+        }
+        response = requests.post(
+            url=f"{base_url}/api/v2/tickets", auth=auth, json=ticket_data
+        )
+        ticket = response.json()["ticket"]
+        yield ticket, user
 
-    x = await graph_task.run_erasure(
-        privacy_request,
-        erasure_policy_string_rewrite,
-        graph,
-        [zendesk_connection_config],
-        {"email": zendesk_erasure_identity_email},
-        get_cached_data_for_erasures(privacy_request.id),
+    @pytest.fixture
+    def zendesk_runner(
+        self,
         db,
-    )
-
-    assert x == {
-        f"{dataset_name}:users": 1,
-        f"{dataset_name}:user_identities": 0,
-        f"{dataset_name}:tickets": 1,
-        f"{dataset_name}:ticket_comments": 0,
-    }
-
-    zendesk_secrets = zendesk_connection_config.secrets
-    auth = zendesk_secrets["username"], zendesk_secrets["api_key"]
-    base_url = f"https://{zendesk_secrets['domain']}"
-
-    # user
-    response = requests.get(
-        url=f"{base_url}/v2/users",
-        auth=auth,
-        params={"email": zendesk_erasure_identity_email},
-    )
-    # Since user is deleted, it won't be available so response is 404
-    assert response.status_code == 404
-
-    for ticket in v[f"{dataset_name}:tickets"]:
-        ticket_id = ticket["id"]
-        response = requests.get(
-            url=f"{base_url}/v2/tickets/{ticket_id}.json",
-            auth=auth,
+        zendesk_secrets,
+    ) -> ConnectorRunner:
+        return ConnectorRunner(db, zendesk_secrets, "zendesk")
+
+    def test_connection(self, zendesk_runner: ConnectorRunner):
+        zendesk_runner.test_connection()
+
+    async def test_access_request(
+        self, zendesk_runner: ConnectorRunner, policy, zendesk_identity_email: str
+    ):
+        access_results = await zendesk_runner.access_request(
+            access_policy=policy, identities={"email": zendesk_identity_email}
         )
-        # Since ticket is deleted, it won't be available so response is 404
-        assert response.status_code == 404
 
-    CONFIG.execution.masking_strict = masking_strict
+    async def test_non_strict_erasure_request(
```

Comment:
Made two separate base methods, one for strict erasure and one for non-strict, so the user doesn't need to worry about setting the strictness config